src/lib/operators/table_scan/sorted_segment_search.hpp

```diff
@@ -34,59 +94,130 @@ class SortedSegmentSearch {
+        return segment_position.value() < value;
       });
     }
   }
 
   // This function sets the offset(s) which delimit the result set based on the predicate condition and the sort order
-  void _set_begin_and_end() {
+  void _set_begin_and_end_positions_for_vs_value_scan() {
     if (_predicate_condition == PredicateCondition::Equals) {
-      _begin = _get_first_bound();
-      _end = _get_last_bound();
+      _begin = _get_first_bound(_first_search_value, _begin, _end);
+      _end = _get_last_bound(_first_search_value, _begin, _end);
       return;
     }
 
     // clang-format off
     if (_is_ascending) {
       switch (_predicate_condition) {
-        case PredicateCondition::GreaterThanEquals: _begin = _get_first_bound(); return;
-        case PredicateCondition::GreaterThan: _begin = _get_last_bound(); return;
-        case PredicateCondition::LessThanEquals: _end = _get_last_bound(); return;
-        case PredicateCondition::LessThan: _end = _get_first_bound(); return;
+        case PredicateCondition::GreaterThanEquals: _begin = _get_first_bound(_first_search_value, _begin, _end); return; // NOLINT
+        case PredicateCondition::GreaterThan: _begin = _get_last_bound(_first_search_value, _begin, _end); return;
+        case PredicateCondition::LessThanEquals: _end = _get_last_bound(_first_search_value, _begin, _end); return;
+        case PredicateCondition::LessThan: _end = _get_first_bound(_first_search_value, _begin, _end); return;
         default: Fail("Unsupported predicate condition encountered");
       }
     } else {
       switch (_predicate_condition) {
-        case PredicateCondition::LessThanEquals: _begin = _get_first_bound(); return;
-        case PredicateCondition::LessThan: _begin = _get_last_bound(); return;
-        case PredicateCondition::GreaterThanEquals: _end = _get_last_bound(); return;
-        case PredicateCondition::GreaterThan: _end = _get_first_bound(); return;
+        case PredicateCondition::LessThanEquals: _begin = _get_first_bound(_first_search_value, _begin, _end); return;
+        case PredicateCondition::LessThan: _begin = _get_last_bound(_first_search_value, _begin, _end); return;
+        case PredicateCondition::GreaterThanEquals: _end = _get_last_bound(_first_search_value, _begin, _end); return;
+        case PredicateCondition::GreaterThan: _end = _get_first_bound(_first_search_value, _begin, _end); return;
         default: Fail("Unsupported predicate condition encountered");
       }
     }
     // clang-format on
   }
 
+  // This function sets the offset(s) that delimit the result set based on the predicate condition and the sort order
+  void _set_begin_and_end_positions_for_between_scan() {
+    DebugAssert(_second_search_value, "Second Search Value must be set for between scan");
+    if (_begin == _end) return;
+
+    auto first_value = _begin->value();
+    auto last_value = (_end - 1)->value();
+    if (!_is_ascending) std::swap(first_value, last_value);
```

Comment:
Please explain what you are doing here. Also, it's probably easier if the whole swapping part is limited to a single block of code (see "exchange predicate conditions and search values if descending")